sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcSourceSuite.scala

```diff
@@ -186,6 +186,54 @@ abstract class OrcSuite extends OrcTest with BeforeAndAfterAll {
     }
   }
 
+  protected def testORCTableLocation(isConvertMetastore: Boolean): Unit = {
+    val tableName1 = "spark_orc1"
+    val tableName2 = "spark_orc2"
+
+    withTempDir { dir =>
+      val someDF1 = Seq((1, 1, "orc1"), (2, 2, "orc2")).toDF("c1", "c2", "c3").repartition(1)
+      withTable(tableName1, tableName2) {
+        val dataDir = s"${dir.getCanonicalPath}/dir1/"
+        val parentDir = s"${dir.getCanonicalPath}/"
+        val wildCardDir = new File(s"${dir}/*").toURI
+        someDF1.write.orc(dataDir)
+        val parentDirStatement =
+          s"""
+             |CREATE EXTERNAL TABLE $tableName1(
+             |  c1 int,
+             |  c2 int,
+             |  c3 string)
+             |STORED AS orc
+             |LOCATION '${parentDir}'""".stripMargin
+        sql(parentDirStatement)
+        val parentDirSqlStatement = s"select * from ${tableName1}"
+        if (isConvertMetastore) {
+          checkAnswer(sql(parentDirSqlStatement), Nil)
+        } else {
+         checkAnswer(sql(parentDirSqlStatement),
+           (1 to 2).map(i => Row(i, i, s"orc$i")))
+        }
+
+        val wildCardStatement =
+          s"""
+             |CREATE EXTERNAL TABLE $tableName2(
+             |  c1 int,
+             |  c2 int,
+             |  c3 string)
+             |STORED AS orc
+             |LOCATION '$wildCardDir'""".stripMargin
```

Comment:
@kevinyu98 . This works, but there is a side effect with this. I mean this creates additional directory who name is '*' literally.

Reply:
@dongjoon-hyun Hello Dongjoon, yes, you are right. It will create a directory with the name is '*', and it is the same behavior prior spark 2.4. I was just following the examples from the jira. Do you have any suggestions here? Thanks.

Reply:
I have two suggestions.

1. Is this PR aiming only one-level subdirectories? Could you check the behavior on one, two, three level subdirectories in Parquet Hive tables first?
2. Since the test case looks general for both Parquet/ORC, please add a test case for Parquet while you are here.

Reply:
@dongjoon-hyun Thanks for the suggestions. I tried with three level subdirectores for Parquet/ORC. Here is the result:

sql("set spark.sql.hive.convertMetastoreOrc=true")

 three level directories

 ORC:

 

 - "/*" can read sub directory data, but not three level subdirectories 
 - "/"  can only read current directory

 Parquet:

 - "/*" can read sub directory data, but not three level subdirectories
 - "/"  can only read current directory
 

 sql("set spark.sql.hive.convertMetastoreOrc=false")

 ORC:

 - "/"  can read three level subdirectories
 - "/*" can't read any data

 parquet:

 - "/"  can only read current directory
 - "/*" can read sub directory data, but not three level subdirectories.

With sql("set spark.sql.hive.convertMetastoreOrc=true"), the ORC and Parquet behavior is consistent. 
1. I think this PR is aiming only one-level subdirectores.
2. Sure, I will add one more for Parquet.

Reply:
Thank you for investigating. I agree with you for (1). For the test case, please add three-level subdirectories. That will help us to improve Spark later. You may file another JIRA issue for that as a new feature JIRA.

Reply:
@dongjoon-hyun Sorry for the delay. My got some issues with my Intellij environment. Sure, I will add three level subdirectories for this PR. FYI, I also tried with `convertMetastoreParquet` for Parquet, the behavior is consistent. 
sql("set spark.sql.hive.convertMetastoreParquet = true")

three level 

Parquet:

-- "/"  can only read current directory
-- "/*" can read sub directory data, but not three level subdirectories.

sql("set spark.sql.hive.convertMetastoreParquet = false")

-- "/"  can only read current directory
-- "/*" can read sub directory data, but not three level subdirectories.

Reply:
@dongjoon-hyun I didn't add three level subdirectores in this PR, should I ? I was thinking to add the three levels in the follow up PR. Let me know what you prefer. Thanks.

Reply:
Thanks, @kevinyu98 . Also, please update the PR title
```
- [Spark-25993][SQL][TEST]Add test cases for resolution of ORC table location 
+ [SPARK-25993][SQL][TEST] Add test cases for CREATE EXTERNAL TABLE with subdirectories
```

Reply:
done